src/ICal/ICal.php

```diff
@@ -2393,31 +2393,27 @@ protected function numberOfDays($days, $start, $end)
      */
     protected function convertDayOrdinalToPositive($dayNumber, $weekday, $timestamp)
     {
-        $dayNumber = empty($dayNumber) ? 1 : $dayNumber; // Returns 0 when no number defined in BYDAY
-
-        $dayOrdinals = $this->dayOrdinals;
-
-        // We only care about negative BYDAY values
-        if ($dayNumber >= 1) {
-            return $dayOrdinals[$dayNumber];
-        }
-
-        $timestamp = (is_object($timestamp)) ? $timestamp : \DateTime::createFromFormat(self::UNIX_FORMAT, $timestamp);
-        $start     = strtotime('first day of ' . $timestamp->format(self::DATE_TIME_FORMAT_PRETTY));
-        $end       = strtotime('last day of ' . $timestamp->format(self::DATE_TIME_FORMAT_PRETTY));
-
-        // Used with pow(2, X) so pow(2, 4) is THURSDAY
-        $weekdays = array_flip(array_keys($this->weekdays));
-
-        $numberOfDays = $this->numberOfDays(pow(2, $weekdays[$weekday]), $start, $end);
-
-        // Create subset
-        $dayOrdinals = array_slice($dayOrdinals, 0, $numberOfDays, true);
-
-        // Reverse only the values
-        $dayOrdinals = array_combine(array_keys($dayOrdinals), array_reverse(array_values($dayOrdinals)));
-
-        return $dayOrdinals[$dayNumber * -1];
+		$dayNumber = empty($dayNumber) ? 1 : $dayNumber; // Returns 0 when no number defined in BYDAY
+
+		$dayOrdinals = $this->dayOrdinals;
+
+		// We only care about negative BYDAY values
+		if ($dayNumber >= 1) {
+			return $dayOrdinals[$dayNumber];
+		}
+
+		$ords = array("", "second", "third", "fourth", "fifth");
+		$weekdays = array("SU"=>"Sunday", "MO"=>"Monday", "TU"=>"Tuesday", "WE"=>"Wednesday", "TH"=>"Thursday", "FR"=>"Friday", "SA"=>"Saturday");
+		$month = date("F", $timestamp);
```

Comment:
What happens when `$timestamp` is a DateTime object?

If you run this against `.\examples\ICal.ics`, you will get:

- `Warning: date() expects parameter 2 to be int, object given`

I would test against this file to make sure your method changes return the same output.